app/scripts/views/components/survivalChart/template.js

```diff
@@ -50,17 +49,94 @@
         this.showLoad = true;
       },
       'update-special-charts': function(hasFilters) {
+        this.updatePlot(hasFilters, this.excludeNa);
+      },
+      'changeExcludeNa': function(excludeNa) {
+        this.updatePlot(this.hasFilters, excludeNa);
+      },
+      'closeChart': function() {
+        this.invisibleDimension.dispose();
+        this.$dispatch('close');
+      },
+      'addingChart': function(groupId, val) {
+        if (this.attributes.group_id === groupId) {
+          if (this.attributes.filter.length > 0) {
+            if (val) {
+              this.invisibleDimension.filterAll();
+            } else {
+              var filtersMap = {};
+              _.each(this.attributes.filter, function(filter) {
+                if (filtersMap[filter] === undefined) {
+                  filtersMap[filter] = true;
+                }
+              });
+              this.invisibleDimension.filterFunction(function(d) {
+                return (filtersMap[d] !== undefined);
+              });
+            }
+          }
+        }
+      },
+      'create-rainbow-survival': function(opts) {
+        var _opts = $.extend(true, {}, opts);
+        _opts.groups = this.calcCurvesData(
+          _opts.groups, _opts.groupType);
+
+        if (_opts.subtitle) {
+          this.displayName = this.attributes.display_name + _opts.subtitle;
+        }
+        this.chartInst.update(
+          _opts.groups, this.chartId, this.attributes.attr_id);
+        this.checkDownloadableStatus();
+        this.updateQtipContent();
+      }
+    },
+    methods: {
+      updatePlot: function(hasFilters, excludeNa) {
         var _type = this.attributes.group_type;
         var attrId = _type === 'patient' ? 'patient_uid' : 'sample_uid';
         var _selectedCases = [];
-        var _allCases = iViz.getCaseUIDs(_type);
+        var _allCases = iViz.getCaseUIDs(_type).sort();
         var groups = [];
 
-        if (hasFilters) {
-          _selectedCases =
-            _.pluck(this.invisibleDimension.top(Infinity), attrId);
+        if (_.isBoolean(excludeNa)) {
+          this.excludeNa = excludeNa;
         }
 
+        if (_.isBoolean(hasFilters)) {
+          this.hasFilters = hasFilters;
+        }
+
+        if (this.hasFilters) {
+          var filteredClinicalAttrs = _.flatten(_.map(this.$root.groups, function(group) {
+            return _.pluck(_.filter(group.attributes, function(attr) {
+              return attr.filter.length > 0;
+            }), 'attr_id');
+          }));
+          if (this.excludeNa) {
+            var data_ = iViz.getGroupNdx(this.attributes.group_id);
+            var _nonNaCases = [];
+            _.each(data_, function(data) {
+              var _intersection = _.intersection(Object.keys(data), filteredClinicalAttrs);
```

Comment:
Yeah, but this intersection should have a very minimium impact. I have used iViz.uti.intersection for other places